Kuroba/app/src/main/java/com/github/adamantcheese/chan/core/loader/impl/external_media_service/ExternalMediaServiceExtraInfoFetcher.kt

```diff
@@ -0,0 +1,33 @@
+package com.github.adamantcheese.chan.core.loader.impl.external_media_service
+
+import com.github.adamantcheese.base.ModularResult
+import com.github.adamantcheese.chan.core.loader.impl.post_comment.LinkInfoRequest
+import com.github.adamantcheese.chan.core.loader.impl.post_comment.SpanUpdateBatch
+import com.github.adamantcheese.database.data.video_service.MediaServiceType
+import io.reactivex.Flowable
+
+/**
+ * Base interface for link extra info fetcher. For now only [YoutubeMediaServiceExtraInfoFetcher] is
+ * supported.
+ * */
+internal interface ExternalMediaServiceExtraInfoFetcher {
+    /**
+     * Each fetcher must have it's own type
+     * */
+    val mediaServiceType: MediaServiceType
+
+    fun isEnabled(): Boolean
+
+    fun fetch(requestUrl: String, linkInfoRequest: LinkInfoRequest): Flowable<ModularResult<SpanUpdateBatch>>
```

Comment:
I think this method should return Single because it's expected to return only one result and Flowable may be used to return a stream of results which would probably break things.